SalesforceCommerceCloud/pwa-kit
# @W-14131330@ Refactor auth storage

# Description

The infinite loop in Storefront Preview in Safari was related to cookie storage, so I made some changes to the auth and storage helpers while investigating. Most of them I subsequently removed, but there were a few things I thought worth keeping.

* Debugging was hard. Using `return await` provides better stack traces than just `return`.
* There were four different classes in `storage.ts`, so I moved them into four different files.
* I added helpers for robust local storage / cookie detection logic. We don't gracefully handle clients with cookies disabled yet, but these helpers are a necessary step for unblocking #1450.



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Refactoring

# How to Test-Drive This PR

- Use the storefront. Auth should work.
- Disable cookies and try again.
- Disable local storage and try again.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
